### PR TITLE
Contracts: Enabled signed extension

### PR DESCRIPTION
**Pull request by pgherveou**
Ideally we fix my other PR #14563 so we can have a test that verify that an Ink contract compiled with 1.70 works properly

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3174096

**Comment by athei** [comment]
> have a test that verify that an Ink contract compiled with 1.70 works properly

Ideally, `pallet-contracts` should not depend on ink! at all or even know about it's existence. Therefore tests should not use ink!. It is enough to write a wat fixture that uses the new instructions. There is no reason to assume that ink! does anything different than emitting those instructions.

**Comment by pgherveou** [comment]
fair I will close the other PR and rebase this one on master

**Review by xermicus — COMMENTED** [review]
How's this tested or do we have a wat fixture for this in master already?

**Comment by pgherveou** [comment]
> How's this tested or do we have a wat fixture for this in master already?

@xermicus  what kind of .wat do we write to proof that we support signed_extension?

**Comment by xermicus** [comment]
Probably something very similar to what [wasm uses to test iti](https://github.com/paritytech/wasmi/blob/master/crates/wasmi/tests/spec/local/missing-features/sign-extension-disabled.wast) should work

```
(module
	(import "env" "memory" (memory 1 1))
	(func (export "deploy"))
	(func (export "call"))
	(func (param i32) (result i32)
		local.get 0
		i32.extend8_s
	)
)
```

File: frame/contracts/src/tests.rs [hunk 1]
```diff
@@ -5633,3 +5633,27 @@ fn root_cannot_instantiate() {
 		);
 	});
 }
+
+#[test]
+fn sign_extension_works() {
+	let (code, _) = compile_module::<Test>("sign_extension").unwrap();
+
+	ExtBuilder::default().build().execute_with(|| {
```
Comment (pgherveou) [inline frame/contracts/src/tests.rs:5641]: this fails with code rejected when sign_extension is false

**Review by xermicus — APPROVED** [review]

File: frame/contracts/fixtures/sign_extension.wat [hunk 2]
```diff
@@ -0,0 +1,9 @@
+(module
```
Comment (agryaznov) [inline frame/contracts/fixtures/sign_extension.wat:1]:
```suggestion
;; Simple fixture which requires sign_extension proposal to be enabled.
(module
```

File: frame/contracts/src/tests.rs (hunk 1 above)
Comment (agryaznov) [reply frame/contracts/src/tests.rs:5641]: how about adding some assertion to the test which also checks that the [sign-extension](https://github.com/WebAssembly/spec/blob/main/proposals/sign-extension-ops/Overview.md) operator (`i32.extend8_s` in your fixture) produces the right value?

**Review by agryaznov — CHANGES_REQUESTED** [review]

**Review by juangirini — COMMENTED** [review]
LGTM I don't give it the green check mark cause of @agryaznov's comment

File: frame/contracts/src/tests.rs (hunk 1 above)
Comment (pgherveou) [reply frame/contracts/src/tests.rs]: actually moved the test to wasm/prepare, seems like a more appropriate place for this test. I don't think we should test the output of the wasm ops (unless you want to be exhaustive and verify all of them)

**Review by agryaznov — APPROVED** [review]

**Comment by pgherveou** [comment]
bot merge

**Pull request merged**